### ISSUE TITLE
X20 Pro - You can no longer select external antenna for 900MHz

### ISSUE THREAD

**Issue by MrD-RC** [FrSkyRC/ETHOS-Feedback-Community#2896]
The option to select the external antenna for 900MHz has been removed. Please put this back. Or allow pilots to select which two frequencies have an external antenna, in the system menu.

I have changed the configuration of my X20 Pro so that 900MHz and 2.4GHz LoRa have external antennas. But FSK does not. Out of all the frequencies, FSK is the least useful to have an external antenna for. Whereas 900MHz is the most important. Especially for pilots flying LR.

What would be really nice, is something like this.
![image](https://github.com/FrSkyRC/ETHOS-Feedback-Community/assets/17590174/f160df58-8062-4d15-9004-13e63800bfe0)
Of course, a confirmation box to make sure that the physical changes have been made would be highly recommended on these options.

**Comment by mawzthefinn** [comment]
The pro has external connections for the 2.4G sections only, thus the removal of the option for 900M

**Comment by MrD-RC** [comment]
> The pro has external connections for the 2.4G sections only, thus the removal of the option for 900M. Adding the 900m option back to support modifications is likely to result in damaged RF sections

Hence the addition of the confirmation box. Maybe even add notices that this is an advanced feature, and not to change it unless you know what you’re doing and have made the necessary hardware modifications. 

Not being able to use an external 900MHz antenna would be a huge setback AFAIC.

**Comment by mawzthefinn** [comment]
I’d argue it’s actually the opposite for the vast majority of users. FSK gets the most benefit from an external antenna because it’s inherently the shortest range and every little bit helps , while 900M gains little benefit from an external at all but the longest range uses which very few do, especially in the primary markets the X20PRO targets (high end fixed wing)

**Comment by MrD-RC** [comment]
In which case, sure the best solution it to allow the pilot to decide.

I’ve never needed to use an external antenna on 2.4. Even when I’ve really pushed the range. They’re just not required for LOS. The internal antennas have at least double the range of what can be seen with the naked eye. 

The 900MHz side on the other hand can massively benefit from an external antenna. Not only can you use higher gain antennas. You can also orient them vertically, which is much more useful with LR. 

I understand the risks of making this too easy. Enabling the external antenna on the module with nothing connected is not a good idea. But if the functionality is not hardware limited, it should be an option. Even if you need to install and use a LUA script or something to make the change. But, the option should be available.

**Comment by bsongis-frsky** [comment]
Needs to decide how to implement it, it shouldn't be too easy to tick the option

**Comment by MrD-RC** [comment]
I agree. There should definitely be some warnings and steps involved to do this. Getting it wrong would not be good for the hardware. But I think it should still be possible. The actual change inside the radio is easier than rotating the gimbals or changing the RTC battery.

**Comment by MrD-RC** [comment]
Stalebot, you are the bane of my life!

**Comment by MrD-RC** [comment]
Seeing as the X20 PRO AW now has 2.4GHz and 900MHz external LoRa antennas. Is the option to select which modules can use the external antennas closer to being implemented. I would like to use the external 900MHz antenna, and I'm not the only one. The hardware is already in place. So this should be pretty simple.

Also, with regard to the hardware dying if no antenna is connected? Is this really the case? I remember the X9D had protection if an antenna was not connected. It would detect it and shut down the RF module to stop damage from occurring. Surely the X20 Pro has this ability too.

**Comment by shellixyz** [comment]
I'm really looking forward to be able to use a 900MHz antenna on this radio as well. The range really sucks with the internal antenna

**Comment by MrD-RC** [comment]
Until this is implemented. I'm using the Pro AW firmware on my Pro. It's working But I still think an option to set the antennas would be a better solution.

**Comment by shellixyz** [comment]
Is it possible to select external antenna for 900MHz now? It has been almost 2 years this issue has been open. Seems like a basic feature.
